# HG changeset patch
# Parent 84ff4c866238da9f627d4b9bab311b759398ebab
# avoing ppa hooks for gmac interface to imporve samba performance

diff --git a/net/core/dev.c b/net/core/dev.c
--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -3734,8 +3734,11 @@ static int __netif_receive_skb(struct sk
 
 #if defined(CONFIG_LTQ_PPA_API_SW_FASTPATH)
 	if(ppa_sw_fastpath_send_hook!=NULL) {
-		if(ppa_sw_fastpath_send_hook(skb) == 0)
-			return NET_RX_SUCCESS;
+		if(skb->dev != NULL && skb->dev->name != NULL) {
+			if((strcmp(skb->dev->name, "gmac5") != 0) && (strcmp(skb->dev->name, "ngmii4") != 0))
+				if(ppa_sw_fastpath_send_hook(skb) == 0)
+					return NET_RX_SUCCESS;
+		}
 	}
 #endif
 
diff --git a/net/ipv4/ip_output.c b/net/ipv4/ip_output.c
--- a/net/ipv4/ip_output.c
+++ b/net/ipv4/ip_output.c
@@ -254,8 +254,10 @@ static inline int ip_finish_output2(stru
     
             flags = 0;  //  post routing
             flags |= CTINFO2DIR(ctinfo) == IP_CT_DIR_ORIGINAL ? PPA_F_SESSION_ORG_DIR : PPA_F_SESSION_REPLY_DIR;
-    
-            ppa_hook_session_add_fn(skb, ct, flags);
+    		if(skb->dev != NULL && skb->dev->name != NULL) {
+				if((strcmp(skb->dev->name, "gmac5") != 0) && (strcmp(skb->dev->name, "ngmii4") != 0))
+	    	        ppa_hook_session_add_fn(skb, ct, flags);
+			}
 #if defined(CONFIG_LANTIQ_IPQOS_CLASS_ACCELERATION_DISABLE)
 	   }
 #endif
diff --git a/net/netfilter/nf_conntrack_core.c b/net/netfilter/nf_conntrack_core.c
--- a/net/netfilter/nf_conntrack_core.c
+++ b/net/netfilter/nf_conntrack_core.c
@@ -1103,8 +1103,11 @@ nf_conntrack_in(struct net *net, u_int8_
     
             flags = PPA_F_BEFORE_NAT_TRANSFORM;
             flags |= CTINFO2DIR(ctinfo) == IP_CT_DIR_ORIGINAL ? PPA_F_SESSION_ORG_DIR : PPA_F_SESSION_REPLY_DIR;
-    
-            ppa_hook_session_add_fn(skb, ct, flags);
+
+			if(skb->dev != NULL && skb->dev->name != NULL) {
+				if((strcmp(skb->dev->name, "gmac5") != 0) && (strcmp(skb->dev->name, "ngmii4") != 0))
+					ppa_hook_session_add_fn(skb, ct, flags);
+			}
         }
 #endif
 
